File: 683.cpp
#include <iostream>
using namespace std;
long grad(long a, long b) {
    if( a == 0 && b == 0) return 1;
    else if( a == 0) return 1;
    else if( b == 0) return 1;
    else if( a%2 == 1 && b%2 == 1 ) return 0;
    else if( a%2 == 0 && b%2 == 0 ) return grad(a-1, b/2) or grad(a/2, b-1);
    else if( a%2 == 1L ) return grad(a-1, b/2);
    else if( b%2 == 1L ) return grad(a/2, b-1);

    return 0;
}

int main() {
  long a, b;
  cin >> a; 
  cin >> b;
  if( grad(a, b) == 1 ) 
    cout << "Yes" << endl;
  else
    cout << "No" << endl;
}

File: 683.kt
import  java.math.BigDecimal as BigDecimal
fun grad(a:BigDecimal, b:BigDecimal):BigDecimal {
  return when  { 
    a == 0.toBigDecimal() && b == 0.toBigDecimal() -> 1.toBigDecimal()
    a == 0.toBigDecimal() -> 1.toBigDecimal()
    a%2.toBigDecimal() == 1.toBigDecimal() && b%2.toBigDecimal() == 1.toBigDecimal() -> 0.toBigDecimal()
    a%2.toBigDecimal() == 0.toBigDecimal() && b%2.toBigDecimal() == 0.toBigDecimal() -> ( grad(a-1.toBigDecimal(), b/2.toBigDecimal()).toLong() or grad(a/2.toBigDecimal(), b-1.toBigDecimal()).toLong() ).toBigDecimal()
    a%2.toBigDecimal() == 1.toBigDecimal() -> grad(a-1.toBigDecimal(), b/2.toBigDecimal())
    b%2.toBigDecimal() == 1.toBigDecimal() -> grad(a/2.toBigDecimal(), b-1.toBigDecimal())
    else -> 0.toBigDecimal()
  }
}
fun main(args:Array<String>) {
  val (a, b) = readLine()!!.split(" ").map(String::toBigDecimal)
  when(grad(a, b)) {
    1.toBigDecimal() -> "Yes"
    else -> "No"
  }.run(::println)
}
